src/CodeGen_C.cpp

```diff
@@ -526,14 +555,22 @@ class CppVector {
         }
     }
 
-    static Vec shuffle(const Vec &a, const int32_t indices[Lanes]) {
+    template<int N, int First, int Second, int... Rest>
+    static void shuffle_impl(Vec &dst, const Vec &src) {
+        dst.elements[N] = src.elements[First < 0 ? 0 : First];
```

Comment:
It might be better to put this in an `if (First >= 0)`, so the compiler is free to ignore the undef elements.